### ISSUE TITLE
Git repo not found when using node as "script" and xvfb-run as "interpreter"

### ISSUE THREAD

**Issue by wiliame** [Unitech/pm2#2571]
I'm starting an APP using the next template:

```javascript
...
"apps": [
    {
      "name": "project-name",
      "interpreter": "xvfb-run",
      "script": "node",
      "args": "--harmony app.js",
      "post_update": [
        "npm install"
      ],
      "env": {
        "DEBUG": "project-name:*"
      }
    }
  ],
...
```

The command used is: `pm2 start package.json` from the directory of the project, where .git is located.

```
/opt/project-name
/opt/project-name/.git
/opt/project-name/app.js
/opt/project-name/package.json
```

#### Expected behaviour

Do git deploys from keymetrics admin.

#### Actual behaviour

All features are working, but pm2 doesn't detects git. I think that is looking at "script" directory. I have tried with `"cwd": "/opt/project-name"` but nothing changed.

![image](http://i.imgur.com/VrxQA2h.png)

#### Steps to reproduce

Set `xvfb-run` as interpreter and `node` as script. 

With `node` as interpreter and `app.js` as script works as expected, but I need xvfb-run and this is the only way I made it work.

#### Software versions used

```
OS         : Debian 8
node.js    : v6.9.2
PM2        : 2.1.6
```

#### PM2 Log output

Command: `tail --lines 50 ~/.pm2/pm2.log`

```
2016-12-13 19:39:42: pid=26106 msg=failed to kill - retrying in 100ms
2016-12-13 19:39:42: pid=26106 msg=failed to kill - retrying in 100ms
2016-12-13 19:39:42: pid=26106 msg=failed to kill - retrying in 100ms
2016-12-13 19:39:42: pid=26106 msg=failed to kill - retrying in 100ms
2016-12-13 19:39:42: pid=26106 msg=failed to kill - retrying in 100ms
2016-12-13 19:39:42: pid=26106 msg=failed to kill - retrying in 100ms
2016-12-13 19:39:42: pid=26106 msg=failed to kill - retrying in 100ms
2016-12-13 19:39:43: pid=26106 msg=failed to kill - retrying in 100ms
2016-12-13 19:39:43: pid=26106 msg=failed to kill - retrying in 100ms
2016-12-13 19:39:43: Process with pid 26106 still alive after 1600ms, sending it SIGKILL now...
2016-12-13 19:39:43: App [project-name] with id [18] and pid [26106], exited with code [0] via signal [SIGKILL]
2016-12-13 19:39:43: pid=26106 msg=process killed
2016-12-13 19:39:43: Starting execution sequence in -fork mode- for app name:project-name id:18
2016-12-13 19:39:43: App name:project-name id:18 online
2016-12-14 06:36:08: Stopping app:project-name id:18
2016-12-14 06:36:09: pid=26705 msg=failed to kill - retrying in 100ms
2016-12-14 06:36:09: pid=26705 msg=failed to kill - retrying in 100ms
2016-12-14 06:36:09: pid=26705 msg=failed to kill - retrying in 100ms
2016-12-14 06:36:09: pid=26705 msg=failed to kill - retrying in 100ms
2016-12-14 06:36:09: pid=26705 msg=failed to kill - retrying in 100ms
2016-12-14 06:36:09: pid=26705 msg=failed to kill - retrying in 100ms
2016-12-14 06:36:09: pid=26705 msg=failed to kill - retrying in 100ms
2016-12-14 06:36:09: pid=26705 msg=failed to kill - retrying in 100ms
2016-12-14 06:36:09: pid=26705 msg=failed to kill - retrying in 100ms
2016-12-14 06:36:10: pid=26705 msg=failed to kill - retrying in 100ms
2016-12-14 06:36:10: pid=26705 msg=failed to kill - retrying in 100ms
2016-12-14 06:36:10: pid=26705 msg=failed to kill - retrying in 100ms
2016-12-14 06:36:10: pid=26705 msg=failed to kill - retrying in 100ms
2016-12-14 06:36:10: pid=26705 msg=failed to kill - retrying in 100ms
2016-12-14 06:36:10: pid=26705 msg=failed to kill - retrying in 100ms
2016-12-14 06:36:10: Process with pid 26705 still alive after 1600ms, sending it SIGKILL now...
2016-12-14 06:36:10: App [project-name] with id [18] and pid [26705], exited with code [0] via signal [SIGKILL]
2016-12-14 06:36:10: pid=26705 msg=process killed
2016-12-14 06:36:28: Starting execution sequence in -fork mode- for app name:project-name id:19
2016-12-14 06:36:28: App name:project-name id:19 online
2016-12-14 06:36:57: Stopping app:project-name id:19
2016-12-14 06:36:57: App [project-name] with id [19] and pid [6109], exited with code [0] via signal [SIGINT]
2016-12-14 06:36:58: pid=6109 msg=process killed
2016-12-14 06:37:03: Starting execution sequence in -fork mode- for app name:project-name id:20
2016-12-14 06:37:03: App name:project-name id:20 online
2016-12-14 06:37:31: Stopping app:project-name id:20
2016-12-14 06:37:31: App [project-name] with id [20] and pid [6319], exited with code [0] via signal [SIGINT]
2016-12-14 06:37:31: pid=6319 msg=process killed
2016-12-14 06:37:45: Starting execution sequence in -fork mode- for app name:app id:21
2016-12-14 06:37:45: App name:app id:21 online
2016-12-14 06:37:50: Stopping app:app id:21
2016-12-14 06:37:50: App [app] with id [21] and pid [6635], exited with code [0] via signal [SIGINT]
2016-12-14 06:37:51: pid=6635 msg=process killed
2016-12-14 06:41:43: Starting execution sequence in -fork mode- for app name:project-name id:22
2016-12-14 06:41:43: App name:project-name id:22 online
```

cf CONTRIBUTING.md for more informations

**Comment by vmarchaud** [comment]
Can you retry to set `cwd` in your process file and make a `pm2 kill` then restart your ecosystem ? (To make sure that the option is saved on the daemon)

**Comment by wiliame** [comment]
Nothing happened.

### New config

```
  "apps": [
    {
      "name": "project-name",
      "interpreter": "xvfb-run",
      "script": "node",
      "args": "--harmony app.js",
      "post_update": [
        "npm install"
      ],
      "cwd": "/opt/project-name",
      "env": {
        "DEBUG": "project-name:*"
      }
    }
  ],
```

### Admin after kill

![image](http://i.imgur.com/3v0eDeV.png)

### Last 50 logs

```
2016-12-14 10:02:05: Stopping app:pm2-server-monit id:0
2016-12-14 10:02:05: App [pm2-server-monit] with id [0] and pid [10168], exited with code [0] via signal [SIGINT]
2016-12-14 10:02:05: [HandleExit] PM2 is being killed, stopping restart procedure...
2016-12-14 10:02:06: pid=10168 msg=process killed
2016-12-14 10:02:06: Stopping app:project-name id:23
2016-12-14 10:02:06: pid=20706 msg=failed to kill - retrying in 100ms
2016-12-14 10:02:06: pid=20706 msg=failed to kill - retrying in 100ms
2016-12-14 10:02:06: pid=20706 msg=failed to kill - retrying in 100ms
2016-12-14 10:02:06: pid=20706 msg=failed to kill - retrying in 100ms
2016-12-14 10:02:06: pid=20706 msg=failed to kill - retrying in 100ms
2016-12-14 10:02:06: pid=20706 msg=failed to kill - retrying in 100ms
2016-12-14 10:02:06: pid=20706 msg=failed to kill - retrying in 100ms
2016-12-14 10:02:06: pid=20706 msg=failed to kill - retrying in 100ms
2016-12-14 10:02:06: pid=20706 msg=failed to kill - retrying in 100ms
2016-12-14 10:02:07: pid=20706 msg=failed to kill - retrying in 100ms
2016-12-14 10:02:07: pid=20706 msg=failed to kill - retrying in 100ms
2016-12-14 10:02:07: pid=20706 msg=failed to kill - retrying in 100ms
2016-12-14 10:02:07: pid=20706 msg=failed to kill - retrying in 100ms
2016-12-14 10:02:07: pid=20706 msg=failed to kill - retrying in 100ms
2016-12-14 10:02:07: pid=20706 msg=failed to kill - retrying in 100ms
2016-12-14 10:02:07: Process with pid 20706 still alive after 1600ms, sending it SIGKILL now...
2016-12-14 10:02:07: App [project-name] with id [23] and pid [20706], exited with code [0] via signal [SIGKILL]
2016-12-14 10:02:07: [HandleExit] PM2 is being killed, stopping restart procedure...
2016-12-14 10:02:07: pid=20706 msg=process killed
2016-12-14 10:02:07: ===============================================================================
2016-12-14 10:02:07: --- Stopping PM2 --------------------------------------------------------------
2016-12-14 10:02:07: Time                 : Wed Dec 14 2016 10:02:07 GMT+0000 (UTC)
2016-12-14 10:02:07: ===============================================================================
2016-12-14 10:02:07: RPC closed
2016-12-14 10:02:07: PUB closed
2016-12-14 10:02:07: PM2 successfully stopped
2016-12-14 10:02:13: ===============================================================================
2016-12-14 10:02:13: --- New PM2 Daemon started ----------------------------------------------------
2016-12-14 10:02:13: Time                 : Wed Dec 14 2016 10:02:13 GMT+0000 (UTC)
2016-12-14 10:02:13: PM2 version          : 2.1.6
2016-12-14 10:02:13: Node.js version      : 6.9.2
2016-12-14 10:02:13: Current arch         : x64
2016-12-14 10:02:13: PM2 home             : /root/.pm2
2016-12-14 10:02:13: PM2 PID file         : /root/.pm2/pm2.pid
2016-12-14 10:02:13: RPC socket file      : /root/.pm2/rpc.sock
2016-12-14 10:02:13: BUS socket file      : /root/.pm2/pub.sock
2016-12-14 10:02:13: Application log path : /root/.pm2/logs
2016-12-14 10:02:13: Process dump file    : /root/.pm2/dump.pm2
2016-12-14 10:02:13: Concurrent actions   : 2
2016-12-14 10:02:13: SIGTERM timeout      : 1600
2016-12-14 10:02:13: ===============================================================================
2016-12-14 10:02:14: Starting execution sequence in -fork mode- for app name:pm2-server-monit id:0
2016-12-14 10:02:14: App name:pm2-server-monit id:0 online
2016-12-14 10:02:20: Starting execution sequence in -fork mode- for app name:project-name id:1
2016-12-14 10:02:20: App name:project-name id:1 online
```

**Comment by vmarchaud** [comment]
Should be fixed in the next release, see https://github.com/Unitech/pm2/commit/4b82c694b6ea52a6ff69ebb2127dd1fd902f0894

**Comment by vmarchaud** [comment]
Published PM2 `2.4.0`
